examples/api/lib/widgets/overlay/overlay_portal.0.dart

```diff
@@ -0,0 +1,71 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// Flutter code sample for OverlayPortal
+
+import 'package:flutter/gestures.dart';
+import 'package:flutter/material.dart';
+
+void main() => runApp(const MyApp());
+
+class MyApp extends StatelessWidget {
+  const MyApp({super.key});
+
+  static const String _title = 'Flutter Code Sample';
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: _title,
+      home: Scaffold(
+        appBar: AppBar(title: const Text('OverlayPortal Example')),
+        body: const Center(child: HoverWidget()),
+      )
+    );
+  }
+}
+
+class HoverWidget extends StatefulWidget {
+  const HoverWidget({super.key});
+
+  @override
+  State<StatefulWidget> createState() => HoverState();
+}
+
+class HoverState extends State<HoverWidget> {
+  bool isHovering = false;
+
+  void _onEnter(PointerEnterEvent event) {
+    setState(() { isHovering = true; });
+  }
+
+  void _onExit(PointerExitEvent event) {
+    setState(() { isHovering = false; });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return DefaultTextStyle(
+      style: DefaultTextStyle.of(context).style.copyWith(fontSize: 50),
+      child: MouseRegion(
+        onEnter: _onEnter,
+        onExit: _onExit,
+        child: OverlayPortal(
+          overlayInfo: OverlayInfo.of(context),
```

Comment:
What is the purpose of this?

Reply:
It's `Overlay.of` but using an `InheirtedWidget` (`Overlay.of` uses `context.findAncestorStateOfType<OverlayState>()` because we didn't have a declarative API that requires the consumer widget to be notified when the overlay changes).

Also the widget that provides `OverlayInfo` lives under `Overlay` so it knows which `OverlayEntry` it is from. We can add more (private) information to `OverlayInfo` when needed.

Reply:
Is there a reason why `OverlayPortal` can't do that itself? I'd expect `OverlayInfo.of(context)` to use the `OverlayPortal`'s `Element` as `context`

Reply:
Most of the time it should I think
I'd at least ewpect this parameter to be optional

Reply:
Defaults to the closest `Overlay` when null? Then if the developer meant to use a different `Overlay` but used the wrong context it will fail silently and show `overlayChild` on the wrong `Overlay`.

Reply:
Personally I have multiple overlays and in `flutter_portal` I implemented some tag-based selection features. Maybe that is a feasible solution as well?

Here is an example with two Portal(Overlay) and two PortalTarget(OverlayEntry) binding to the portal that it likes: https://github.com/fzyzcjy/flutter_portal/blob/93d93badf41a13b7982c8f7cdd753674fab73a8b/test/widget_test.dart#L1908

Reply:
If the developer wants to have a candidate list of target `Overlay`s with priorities they can do `OverInfo.of(context1) ?? OverlayInfo.of(context2)` right?

Reply:
Using tags is a pretty interesting idea and the main benefit seems to be allowing a "blind" Overlay lookup without having to figure out what the correct `BuildContext` is when there's nested overlays? But from the looks of it it doesn't have to be in this patch, it can made into a wrapper class? `InheritedWidget.of` is an established convention at this point and I think it works well enough in this scenario.

Reply:
If we use `null` for default then unfortunately it can have 2 meanings: either the API user wants to use the default, or we failed to get the `OverlayInfo` from the given `BuildContext` at runtime. You can probably solve this by creating some const value to represent the default (the closest `OverlayInfo`), but I think it's better to keep the API simple and stupid in this case.

Reply:
I wouldn't expect OverlayInfo.of to return null if there's no overlay anyway. I'd expect an exception
So this wouldn't have 2 meanings

Reply:
Yeah usually we have 2 versions for this kind of things, `maybeOf` and `of`.  I'll rename `Overlay.of` to `Overlay.maybeOf` to follow the naming convention.

But I'm a bit hesitant to only expose the version that throws, especially since this is a query API. It will be a bit awkward for people to do something that could otherwise be really simple to achieve: `OverlayInfo.maybeOf(outerContext) ?? Overlay.maybeOf(innerContext)`.

Reply:
By popular demand I'll rename the current default constructor and introduce a new default constructor that doesn't take an `OverlayInfo?`.